content/en/blog/2022/apisix.md

```diff
@@ -0,0 +1,389 @@
+not be displayed.
+
+## Example
+
+This scenario example deploys Collector, Jaeger, and Zipkin as backend services
+by simply modifying the official example of OpenTelemetry Collector, and starts
+two sample applications (Client and Server), where Server provides an HTTP
+service, and Client will cyclically call the server provided by the server. HTTP
+interface, resulting in a call chain consisting of two spans.
+
+### Step 1: Deploy OpenTelemetry
+
+The following uses docker-compose as an example. For other deployments, please
+refer to the OpenTelemetry official documentation.
+
+You can refer to the following command to deploy:
+
+```shell
+git clone https://github.com/open-telemetry/opentelemetry-collector-contrib.git
+cd opentelemetry-collector-contrib/examples/demo
+docker-compose up -d
+```
+
+Enter `http://127.0.0.1:16886` (Jaeger UI) or `http://127.0.0.1:9411/zipkin`
+(Zipkin UI) in the browser. If it can be accessed normally, the deployment is
+successful.
+
+The following screenshots show an example of successful access.
+
+![Jaeger UI](/img/apisix/jaeger-ui-2.png)
+
+![Zipkin UI](/img/apisix/zipkin-ui-1.png)
+
+### Step 2: Configure the Test Environment
+
+The Apache APISIX service is introduced, and the topology of the final
+application is shown in the following figure.
+
+![topology of the demo](/img/apisix/demo-topology.png)
+
+The Trace data reporting process is as follows. Among them, since Apache APISIX
+is deployed separately and not in the network of docker-compose, Apache APISIX
+accesses the OTLP HTTP Receiver of OpenTelemetery Collector through the locally
+mapped port (`127.0.0.1:4138`).
+
+![Trace data reporting process](/img/apisix/trace-data-flow.png)
+
+You need to make sure you have enabled the `opentelemetry` plugin and reload
+Apache APISIX.
+
+You can refer to the following example to create a route and enable
+the`opentelemetry` plugin for sampling:
+  
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 \
+-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' \
+-X PUT -d '
+{
+  "uri": "/hello",
+  "plugins": {
+      "opentelemetry": {
+          "sampler": {
+            "name": "always_on"
+          }
+      }
+  },
+  "upstream": {
+      "type": "roundrobin",
+      "nodes": {
+          "127.0.0.1:7080": 1
+      }
+  }
+}'
+```
+
+Modify the `./examples/demo/otel-collector-config.yaml` file to add the OTLP
+HTTP Receiver.  
+
+```shell
+  receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:${ip:port}   # add OTLP HTTP Receiver，default port is 4318
+```
+
+Modify `docker-compose.yaml` file.
+
+You need to modify the configuration file, change the interface address of
+Client calling Server to the address of Apache APISIX, and map the ports of
+OTLP HTTP Receiver and Server services to local.
+
+The following example is the complete `docker-compose.yaml` after the
+configuration is modified:
+
+  ```yaml
+  version: "2"
+  services:
```

Comment:
Seems we don't need the indent here, right ?

Reply:
Yeah, updated.